File: lesson4/python3/homework/homework4_12.py
#!/usr/bin/python3
#: lesson4/python3/homework/homework4_12.py

numb = int(input("Введите натуральное число => "))

numb_i = 1
while numb_i < numb:
    numb_pow = numb_i ** 2
    if numb_pow % (10 ** 1) == numb_i:
        print("Натуральное число в первой степень =>", numb_i)
    elif numb_pow % (10 ** 2) == numb_i:
        print("Натуральное число в второй степень =>", numb_i)
    elif numb_pow % (10 ** 3) == numb_i:
        print("Натуральное число в третей степень =>", numb_i)
    elif numb_pow % (10 ** 4) == numb_i:
        print("Натуральное число в четвертой степень =>", numb_i)
    elif numb_pow % (10 ** 5) == numb_i:
        print("Натуральное число в пятой степень =>", numb_i)
    elif numb_pow % (10 ** 6) == numb_i:
        print('Натуральное число в шестой степень =>', numb_i)
    elif numb_pow % (10 ** 7) == numb_i:
        print("Натуральное число в седьмой степень =>", numb_i)
    elif numb_pow % (10 ** 8) == numb_i:
        print("Натуральное число в восьмой степень =>", numb_i)
    elif numb_pow % (10 ** 9) == numb_i:
        print("Натуральное число в девятой степень =>", numb_i)
    else:
        if numb_pow % (10 ** 10) == numb_i:
            print("Натуральное число в десятой степень =>", numb_i)
    numb_i += 1

File: lesson4/c_c++/homework/homework4_12.cpp
//: lesson4/c_c++/homework/homework4_12.cpp
#include <iostream>
#include <cmath>

int main() {
    using namespace std;
    int64_t numb;
    cout << "Введите натуральное число => ";
    cin >> numb;

    int64_t numb_i = 1;
    while (numb_i < numb) {
        int64_t numb_pow = numb_i * numb_i;
        if (numb_pow % 10 == numb_i) {
            cout << "Натуральное число в первую степень => " << numb_i << endl;
        } else if (numb_pow % 100 == numb_i) {
            cout << "Натуральное число в второй степень => " << numb_i << endl;
        } else if (numb_pow % 1000 == numb_i) {
            cout << "Натуральное число в третей степень => " << numb_i << endl;
        } else if (numb_pow % 10000 == numb_i) {
            cout << "Натуральное число в червертой степень => " << numb_i << endl;
        } else if (numb_pow % 100000 == numb_i) {
            cout << "Натуральное число в пятой степень => " << numb_i << endl;
        } else if (numb_pow % 1000000 == numb_i) {
            cout << "Натуральное число в шестой степень => " << numb_i << endl;
        } else if (numb_pow % 10000000 == numb_i) {
            cout << "Натуральное число в седьмой степень => " << numb_i << endl;
        } else if (numb_pow % 100000000 == numb_i) {
            cout << "Натуральное число в восьмой степень => " << numb_i << endl;
        } else if (numb_pow % 1000000000 == numb_i) {
            cout << "Натуральное число в девятой степень => " << numb_i << endl;
        } else {
            if (numb_pow % 10000000000 == numb_i) {
                cout << "Натуральные число в десятой степень => " << numb_i << endl;
            }
        }
        numb_i++;
    }
    return 0;
}
